meta-facebook/yv35-rf/src/platform/plat_sensor_table.c

```diff
@@ -167,4 +200,189 @@ sensor_cfg plat_sensor_config[] = {
+	  SENSOR_INIT_STATUS, pre_vr_read, &vr_page_select[0], NULL, NULL, NULL },
+	{ SENSOR_NUM_CUR_VR0V8A, sensor_dev_xdpe12284c, I2C_BUS10, VR_A0V8_ADDR, SMBUS_CUR_CMD,
+	  vr_access, 0, 0, SAMPLE_COUNT_DEFAULT, POLL_TIME_DEFAULT, ENABLE_SENSOR_POLLING, 0,
+	  SENSOR_INIT_STATUS, pre_vr_read, &vr_page_select[1], NULL, NULL, NULL },
+	{ SENSOR_NUM_CUR_VR0V8D, sensor_dev_xdpe12284c, I2C_BUS10, VR_D0V8_ADDR, SMBUS_CUR_CMD,
+	  vr_access, 0, 0, SAMPLE_COUNT_DEFAULT, POLL_TIME_DEFAULT, ENABLE_SENSOR_POLLING, 0,
+	  SENSOR_INIT_STATUS, pre_vr_read, &vr_page_select[0], NULL, NULL, NULL },
+	{ SENSOR_NUM_CUR_VRVDDQAB, sensor_dev_xdpe12284c, I2C_BUS10, VR_VDDQAB_ADDR, SMBUS_CUR_CMD,
+	  vr_access, 0, 0, SAMPLE_COUNT_DEFAULT, POLL_TIME_DEFAULT, ENABLE_SENSOR_POLLING, 0,
+	  SENSOR_INIT_STATUS, pre_vr_read, &vr_page_select[1], NULL, NULL, NULL },
+	{ SENSOR_NUM_CUR_VRVDDQCD, sensor_dev_xdpe12284c, I2C_BUS10, VR_VDDQCD_ADDR, SMBUS_CUR_CMD,
+	  vr_access, 0, 0, SAMPLE_COUNT_DEFAULT, POLL_TIME_DEFAULT, ENABLE_SENSOR_POLLING, 0,
+	  SENSOR_INIT_STATUS, pre_vr_read, &vr_page_select[0], NULL, NULL, NULL },
+
+	// VR Power
+	{ SENSOR_NUM_PWR_VR0V9A, sensor_dev_xdpe12284c, I2C_BUS10, VR_A0V9_ADDR, SMBUS_PWR_CMD,
+	  vr_access, 0, 0, SAMPLE_COUNT_DEFAULT, POLL_TIME_DEFAULT, ENABLE_SENSOR_POLLING, 0,
+	  SENSOR_INIT_STATUS, pre_vr_read, &vr_page_select[0], NULL, NULL, NULL },
+	{ SENSOR_NUM_PWR_VR0V8A, sensor_dev_xdpe12284c, I2C_BUS10, VR_A0V8_ADDR, SMBUS_PWR_CMD,
+	  vr_access, 0, 0, SAMPLE_COUNT_DEFAULT, POLL_TIME_DEFAULT, ENABLE_SENSOR_POLLING, 0,
+	  SENSOR_INIT_STATUS, pre_vr_read, &vr_page_select[1], NULL, NULL, NULL },
+	{ SENSOR_NUM_PWR_VR0V8D, sensor_dev_xdpe12284c, I2C_BUS10, VR_D0V8_ADDR, SMBUS_PWR_CMD,
+	  vr_access, 0, 0, SAMPLE_COUNT_DEFAULT, POLL_TIME_DEFAULT, ENABLE_SENSOR_POLLING, 0,
+	  SENSOR_INIT_STATUS, pre_vr_read, &vr_page_select[0], NULL, NULL, NULL },
+	{ SENSOR_NUM_PWR_VRVDDQAB, sensor_dev_xdpe12284c, I2C_BUS10, VR_VDDQAB_ADDR, SMBUS_PWR_CMD,
+	  vr_access, 0, 0, SAMPLE_COUNT_DEFAULT, POLL_TIME_DEFAULT, ENABLE_SENSOR_POLLING, 0,
+	  SENSOR_INIT_STATUS, pre_vr_read, &vr_page_select[1], NULL, NULL, NULL },
+	{ SENSOR_NUM_PWR_VRVDDQCD, sensor_dev_xdpe12284c, I2C_BUS10, VR_VDDQAB_ADDR, SMBUS_PWR_CMD,
+	  vr_access, 0, 0, SAMPLE_COUNT_DEFAULT, POLL_TIME_DEFAULT, ENABLE_SENSOR_POLLING, 0,
+	  SENSOR_INIT_STATUS, pre_vr_read, &vr_page_select[0], NULL, NULL, NULL },
+};
+
+int check_pwr_monitor_type(void)
+{
+	uint8_t retry = 5;
+	uint8_t ret = 0;
+	I2C_MSG msg = { 0 };
+	msg.bus = I2C_BUS1;
+	msg.target_addr = INA233_12V_ADDR;
+	msg.tx_len = 1;
+	msg.rx_len = 7;
+	msg.data[0] = PMBUS_MFR_ID;
+
+	if (i2c_master_read(&msg, retry)) {
+		printf("Failed to read Power moniter IC_DEVICE_ID: register(0x%x)\n",
+		       PMBUS_IC_DEVICE_ID);
+		return -1;
+	}
+
+	if (memcmp(msg.data, INA233_DEVICE_ID, sizeof(INA233_DEVICE_ID)) == 0) {
+		printf("Power Monitor type: INA233\n");
+		ret = PWR_INA233;
+	} else {
```

Comment:
Is it a safe assumption that if the power monitor isn't the INA233, it will always be SGY?  If a future third power monitor is added, it could cause issues.  Suggestion would be to test for each power monitor type and have an else clause to catch if an unknown power monitor is encountered.

Reply:
**EVT RFB**
- BOM1: ina233 + MainVR
- BOM2: ina233 + 2ndVR
- BOM3: SGY_SQ5220x+ MainVR

**DVT RFB** (Support using GPIO to check the type of power monitors)
- BOM1: ina233+MainVR
- BOM2: SGY_SQ5220x+2ndVR
- BOM3: ina233+MainVR

This assumption is just for the case of **EVT RFB** because BIC cannot get the device ID from the SGY_SQ5220x power monitor.
The future work will distinguish EVT RFB and DVT RFB by the board revision, then use GPIO to get the type of power monitor( ina233 or SGY_SQ5220x) in the case of **DVT RFB**.